### PR TITLE
Merge pull request #1 from christianbundy/master

### PR DESCRIPTION
**Pull request by cbleslie**
So... I changed:  
- state management to redux, this is the biggest one; it avoids a lot of mutation to the chat state.
- ensured that most of the functions are referentialy transparent. Just to try an get rid of side-effects that could happen later.
- a bunch of house keeping

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,10 +1,12 @@
 {
-  "name": "hello-express",
+  "name": "html-chat",
   "version": "0.0.1",
   "description": "A live-updating chat room without any browser JavaScript",
   "main": "server.js",
   "scripts": {
-    "start": "node -r esm server.js"
+    "start": "node -r esm server.js",
```
Comment (cbleslie) [inline package.json:7]: using esm to get es6 imports working.

File: package.json [hunk 2]
```diff
@@ -20,10 +22,17 @@
...
+    "chat"
   ],
   "devDependencies": {
-    "eslint-config-node": "^4.0.0"
-  }
+    "eslint-config-node": "^4.0.0",
+    "common-good": "^2.0.3",
+    "stylelint-config-recommended": "^3.0.0"
+  },
+  "author": "Christian Bundy",
```
Comment (cbleslie) [inline package.json]: I could have my name here... someday :P

**Comment by christianbundy** [comment]
:shipit:

File: package.json (hunk 2 above)
Comment (christianbundy) [reply package.json]: I'd 100% merge a patch turning that into `contributors` instead of `author`.

**Pull request merged**